### PR TITLE
[FIX] stock_landed_costs: fix allowed pickings

### PR DESCRIPTION
**Pull request by fja-odoo**
The landed cost journal (lc_journal_id) is not required by default as a
result it is possible to create a landed cost with no account_journal_id
and thus no company_id this will make the query in
_compute_allowed_picking_ids fail.

Issue caused by: #63742

Description of the issue/feature this PR addresses:
Install stock_landed_costs
go to inventory module > operations > landed costs > create

Current behavior before PR:
The query in _compute_allowed_picking_ids fails

Desired behavior after PR is merged:
A landed cost record is created

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock_landed_costs/models/stock_landed_cost.py [hunk 1]
```diff
@@ -82,16 +82,17 @@ def _compute_total_amount(self):
...
-                           INNER JOIN stock_valuation_layer AS svl ON svl.stock_move_id = sm.id
-                                WHERE sm.picking_id IS NOT NULL AND sm.company_id IN %s
-                             GROUP BY sm.picking_id, sm.company_id""", [tuple(self.company_id.ids)])
         valued_picking_ids_per_company = defaultdict(list)
-        for res in self.env.cr.fetchall():
-            valued_picking_ids_per_company[res[1]].append(res[0])
+        if self.company_id:
+            self.env.cr.execute("""SELECT sm.picking_id, sm.company_id
+                                    FROM stock_move AS sm
+                            INNER JOIN stock_valuation_layer AS svl ON svl.stock_move_id = sm.id
```
Comment (ryv-odoo) [inline addons/stock_landed_costs/models/stock_landed_cost.py:89]: Maybe avoid break my beautiful indent :cry:

File: addons/stock_landed_costs/models/stock_landed_cost.py [hunk 2]
```diff
@@ -82,16 +82,17 @@ def _compute_total_amount(self):
...
+                                    WHERE sm.picking_id IS NOT NULL AND sm.company_id IN %s
+                                GROUP BY sm.picking_id, sm.company_id""", [tuple(self.company_id.ids)])
+            for res in self.env.cr.fetchall():
+                valued_picking_ids_per_company[res[1]].append(res[0])
         for cost in self:
-            cost.allowed_picking_ids = valued_picking_ids_per_company[cost.company_id.id]
+            cost.allowed_picking_ids = valued_picking_ids_per_company.get(cost.company_id.id)
```
Comment (ryv-odoo) [inline addons/stock_landed_costs/models/stock_landed_cost.py:95]: Why change this, the defaultdict return a empty list (instead of `None` in your case).

**Review by ryv-odoo — COMMENTED** [review]
Outch my bad! Thank you to fix it.

File: addons/stock_landed_costs/models/stock_landed_cost.py (hunk 2 above)
Comment (fja-odoo) [reply addons/stock_landed_costs/models/stock_landed_cost.py]: Right, I should start using defaultdict !!

**Comment by ryv-odoo** [comment]
LGTM, Thanks.
@robodoo r+

**Pull request closed without merging**